monotonic_arena.go

```diff
@@ -26,8 +26,8 @@ func (s *monotonicBuffer) alloc(size, alignment uintptr) (unsafe.Pointer, bool)
 		s.ptr = unsafe.Pointer(unsafe.SliceData(buf))
 	}
 	alignOffset := uintptr(0)
-	for alignedPtr := uintptr(s.ptr) + s.offset; alignedPtr%alignment != 0; alignedPtr++ {
-		alignOffset++
+	if delta := (uintptr(s.ptr) + s.offset) % alignment; delta != 0 {
+		alignOffset = delta
```

Comment:
this is wrong, and will not result in an aligned address.

Reply:
@mumbleskates is right.

Reply:
yeah, absolutely right, but completely unuseful.

@ortuman those lines are updated.